jax/_src/scipy/special.py

```diff
@@ -665,14 +665,19 @@ def rel_entr(
   """
   p, q = promote_args_inexact("rel_entr", p, q)
   zero = _lax_const(p, 0.0)
-  both_gt_zero_mask = lax.bitwise_and(lax.gt(p, zero), lax.gt(q, zero))
-  one_zero_mask = lax.bitwise_and(lax.eq(p, zero), lax.ge(q, zero))
+
+  # if p is negative, set q to zero to produce output of xlogy to -inf
+  otherwise_mask = lax.bitwise_or(lax.lt(p, zero), lax.lt(q, zero))
+  otherwise_mask = lax.bitwise_or(otherwise_mask, lax.bitwise_and(lax.eq(q, zero), lax.ne(p, zero)))
+  safe_p = jnp.where(otherwise_mask, 1, p)
+  safe_q = jnp.where(otherwise_mask, 0, q)
+  return lax.sub(_xlogx(safe_p), xlogy(safe_p, safe_q))
```

Comment:
Please delete all this new code – it's not necessary as far as I can tell.

All we need is to change `q` to `zero`, as you've done below.

Reply:
I've deleted it.